### ISSUE TITLE
Inconsisten behavior around uncomfortable previous variable names [TW: previous version of JH variable names]

### ISSUE THREAD

**Issue by distortedsignal** [jupyterhub/jupyterhub#3736]
### Bug description
There seems to be inconsistent handling of the words `whitelist` and `blacklist` (previous version JH variable names) when applied to new variable names `allowed_users` and `blocked_users`. Setting `whitelist` appears to set the `allowed_users` variable, while setting the `blacklist` variable does not set the `blocked_users` variable. This is somewhat inconvenient and confusing.

#### Expected behaviour
I would expect these (outdated) terms to have consistent behavior - either setting one (outdated) term would set the (in-fashion) new variable, or neither variable would be set.

#### Actual behaviour
I think it might be easiest to see what I'm talking about in a video.

Setting `whitelist` also sets `allowed_users`.

https://user-images.githubusercontent.com/2186874/147146705-525c5d39-5718-4daf-83f2-a23f367ef9dc.mov

Setting `blacklist` DOES NOT set `blocked_users`.

https://user-images.githubusercontent.com/2186874/147146773-8f1e25b7-f3ed-4b90-b2b2-2b872a6bc20d.mov

### How to reproduce

1. Start JupyterHub with a `whitelist` and a `blacklist` set on the authenticator.
2. Check for the `blocked_users` and `allowed_users` variables on the authenticator.
3. EXPECTED: EITHER BOTH the `blocked_users` and `allowed_users` should be set, or NEITHER should be set.
4. ACTUAL: In my experience, the `allowed_users` var is set, while the `blocked_users` var is not set.

### Your personal set up

 - OS: Manjaro Linux, Kernel Version 5.14.21-2-MANJARO (64-bit), Qt Version: 5.15.2, KDE Plasma Version 5.23.4
 
 - Version(s): jupyterhub version 2.0.1, python version 3.9.9
 

- <details><summary>Full environment</summary>

```
# output of `pip freeze`:
alembic==1.7.5
async-generator==1.10
attrs==21.2.0
certifi==2021.10.8
certipy==0.1.3
cffi==1.15.0
charset-normalizer==2.0.9
coverage==6.2
cryptography==36.0.1
eight==1.0.1
entrypoints==0.3
future==0.18.2
greenlet==1.1.2
idna==3.3
iniconfig==1.1.1
Jinja2==3.0.3
jsonschema==4.3.2
jupyter-telemetry==0.1.0
jupyterhub==2.0.1
lxml==4.7.1
Mako==1.1.6
MarkupSafe==2.0.1
oauthlib==3.1.1
packaging==21.3
pamela==1.0.0
pluggy==1.0.0
prometheus-client==0.12.0
py==1.11.0
pycparser==2.21
pyOpenSSL==20.0.1
pyparsing==3.0.6
pyrsistent==0.18.0
pytest==6.2.5
pytest-asyncio==0.16.0
pytest-cov==3.0.0
python-dateutil==2.8.2
python-json-logger==2.0.2
pytz==2021.3
requests==2.26.0
ruamel.yaml==0.17.17
ruamel.yaml.clib==0.2.6
signxml==2.9.0
six==1.16.0
SQLAlchemy==1.4.28
toml==0.10.2
tomli==2.0.0
tornado==6.1
traitlets==5.1.1
urllib3==1.26.7
```
</details>

- <details><summary>Configuration</summary>

This isn't actually a live jupyter instance, this is unit tests of an authenticator, specifically the JupyterHub-SamlAuthenticator.

```python
# jupyterhub_config.py
```
</details>

**Comment by distortedsignal** [comment]
The last time that I had issues relating to this code, @consideRatio and @minrk were able to help out, so I'm tagging them here.

**Comment by distortedsignal** [comment]
Bump to prevent this issue going stale.

**Comment by minrk** [comment]
No need to bump after only a few days, there's not a lot of activity over the holiday.

You're right that the deprecated `blacklist` has not worked since it was deprecated. But since it's been so long (over a year since 1.2), I'm tempted to _not_ re-add the deprecated functionality, and make the accidental removal official and permanent.

**Comment by distortedsignal** [comment]
@minrk - that's fine. I completely agree with that. I would prefer that the same behavior be applied to the `whitelist` variable, since the distinction between the two is what is confusing me. If we're not supporting setting `blacklist`, then we should not be supporting setting `whitelist`. I understand that this will result in breaking changes for some users, but it will make developing authenticators etc. much easier.

**Comment by minrk** [comment]
Yeah, I get that. If I'd known the deprecation wasn't working quite right, I would have dropped support for the deprecated names in 2.0. The distinction is definitely not intentional, it's just a bug.

Can I ask what the complication is?

```python
if jupyterhub.version_info >= (1, 2):
    use_new_name
else:
    use_old_name
```

ought to work fine. Or, if it's an option available to you, make the next release of your authenticator require jupyterhub 1.2.

Plus, Authenticators shouldn't usually need to call their own `check_allowed_users` and friends, since that's done in JupyterHub itself. The handling of the deprecated names ought to be taken care of by JupyterHub itself.
